lib/Analysis/BlockFrequencyInfoImpl.cpp

```diff
@@ -549,6 +550,8 @@ BlockFrequencyInfoImplBase::getProfileCountFromFreq(const Function &F,
   APInt BlockFreq(128, Freq);
   APInt EntryFreq(128, getEntryFreq());
   BlockCount *= BlockFreq;
+  BlockCount +=
```

Comment:
EntryFreq is just uint64 wrapped by an APInt. It is more efficient to assign getEntryFreq() above (a uint64_t) to a temp, then divide that by 2 and add id to BlockCount.